### PR TITLE
Adds a Playlists vertical and a Spotify adapter

### PR DESCRIPTION
**Pull request by bwillard**
Starts to work on #495

### CODE REVIEW

File: extensions/data-transfer/portability-data-transfer-spotify/Readme.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Mastodon
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-spotify/Readme.md]: Fix name

File: portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java [hunk 2]
```diff
@@ -140,13 +140,17 @@ public void poll() {
     worker.doWork();
   }
 
-  private static List<TransferExtension> getTransferExtensions() {
+  private static List<TransferExtension> getTransferExtensions(Monitor monitor) {
```
Comment (holachuy) [inline portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java]: Merge error?

File: portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java [hunk 3]
```diff
@@ -140,13 +140,17 @@ public void poll() {
...
     // TODO: Next version should ideally not load every TransferExtension impl, look into
     // solutions where we selectively invoke class loader.
     ImmutableList.Builder<TransferExtension> extensionsBuilder = ImmutableList.builder();
     // Note that initialization of the TransferExtension is done in the WorkerModule since they're
     // initialized as they're requested.
-    ServiceLoader.load(TransferExtension.class).iterator().forEachRemaining(extensionsBuilder::add);
+    ServiceLoader.load(TransferExtension.class).iterator()
```
Comment (holachuy) [inline portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java]: merge error?

File: portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerModule.java [hunk 4]
```diff
@@ -23,29 +25,26 @@
...
 import org.datatransferproject.api.launcher.Monitor;
 import org.datatransferproject.config.FlagBindingModule;
 import org.datatransferproject.security.AsymmetricKeyGenerator;
 import org.datatransferproject.security.SymmetricKeyGenerator;
 import org.datatransferproject.spi.cloud.extension.CloudExtension;
-import org.datatransferproject.spi.transfer.hooks.JobHooks;
 import org.datatransferproject.spi.cloud.storage.AppCredentialStore;
 import org.datatransferproject.spi.cloud.storage.JobStore;
 import org.datatransferproject.spi.transfer.extension.TransferExtension;
+import org.datatransferproject.spi.transfer.hooks.JobHooks;
```
Comment (holachuy) [inline portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerModule.java]: merge error?

File: portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java (hunk 2 above)
Comment (bwillard) [reply portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java]: No this is intentional, I had a bug where the plugin wasn't loading so I added this logging to help.

File: portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerModule.java (hunk 4 above)
Comment (bwillard) [reply portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerModule.java]: No I fixed the import order.

**Comment by bwillard** [comment]
PTAL?

**Pull request merged**